<a href="https://colab.research.google.com/github/greyga/TFM-GRG/blob/main/COMPARACIONES_DATOS_genes.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# Crear el DataFrame "datos_Asthma"
datos_Asthma <- data.frame(
  Variante = c("PTGER2", "TBX21", "HNMT", "ADRB2", "PLA2G7"),
  Ranking = c(1, 2, 8, 18, 21),
  Phenotypic_Score = c(0.92, 0.92, 0.50, 0.00, 0.50),
  stringsAsFactors = FALSE
)

# Imprimir el DataFrame "datos_Asthma"
print(datos_Asthma)

  Variante Ranking Phenotypic_Score
1   PTGER2       1             0.92
2    TBX21       2             0.92
3     HNMT       8             0.50
4    ADRB2      18             0.00
5   PLA2G7      21             0.50


In [ ]:
str(datos_Asthma)

'data.frame':	5 obs. of  3 variables:
 $ Variante        : chr  "PTGER2" "TBX21" "HNMT" "ADRB2" ...
 $ Ranking         : num  1 2 8 18 21
 $ Phenotypic_Score: num  0.92 0.92 0.5 0 0.5


In [ ]:
# Crear el DataFrame "datos_Sin_HPO"
datos_Sin_HPO <- data.frame(
  Variante = c("PTGER2", "TBX21", "HNMT", "ADRB2", "PLA2G7"),
  Ranking = c(3, 4, 10, 11, 21),
  Phenotypic_Score = c (0.0, 0.0 ,0.0 , 0.0 ,0.0),
  stringsAsFactors = FALSE
)

# Imprimir el DataFrame "datos_Sin_HPO"
print(datos_Sin_HPO)


  Variante Ranking Phenotypic_Score
1   PTGER2       3                0
2    TBX21       4                0
3     HNMT      10                0
4    ADRB2      11                0
5   PLA2G7      21                0


In [ ]:
# Crear el DataFrame "datos_Asthma_ARS"
datos_Asthma_ARS <- data.frame(
  Variante = c("PTGER2", "TBX21", "HNMT", "ADRB2", "PLA2G7"),
  Ranking = c(1, 2, 8, 18, 21),
  Phenotypic_Score = c(1, 1, 0.5, 0.0, 0.5),
  stringsAsFactors = FALSE
)

# Imprimir el DataFrame "datos_Asthma_ARS"
print(datos_Asthma_ARS)


  Variante Ranking Phenotypic_Score
1   PTGER2       1              1.0
2    TBX21       2              1.0
3     HNMT       8              0.5
4    ADRB2      18              0.0
5   PLA2G7      21              0.5


In [ ]:
# Crear el DataFrame "datos_Asthma_ARP"
datos_Asthma_ARP <- data.frame(
  Variante = c("PTGER2", "TBX21", "HNMT", "ADRB2", "PLA2G7"),
  Ranking = c(1, 2, 8, 18, 21),
  Phenotypic_Score = c(1, 1, 0.5, 0.0, 0.5),
  stringsAsFactors = FALSE
)

# Imprimir el DataFrame "datos_Asthma_ARP"
print(datos_Asthma_ARP)


  Variante Ranking Phenotypic_Score
1   PTGER2       1              1.0
2    TBX21       2              1.0
3     HNMT       8              0.5
4    ADRB2      18              0.0
5   PLA2G7      21              0.5


In [ ]:
# Crear el DataFrame "datos_Ab_Resp_system"
datos_Ab_Resp_system <- data.frame(
  Variante = c("PTGER2", "TBX21", "HNMT", "ADRB2", "PLA2G7"),
  Ranking = c(1, 2, 8, 16, 21),
  Phenotypic_Score = c(1, 1, 0.5, 0.0, 0.5),
  stringsAsFactors = FALSE
)

# Imprimir el DataFrame "datos_Ab_Resp_system"
print(datos_Ab_Resp_system)


  Variante Ranking Phenotypic_Score
1   PTGER2       1              1.0
2    TBX21       2              1.0
3     HNMT       8              0.5
4    ADRB2      16              0.0
5   PLA2G7      21              0.5


In [ ]:
# Crear el DataFrame "datos_Asthma_Ig_hypers"
datos_Asthma_Ig_hypers <- data.frame(
  Variante = c("PTGER2", "TBX21", "HNMT", "ADRB2", "PLA2G7"),
  Ranking = c(1, 2, 8, 16, 21),
  Phenotypic_Score = c(0.93, 0.93, 0.5, 0.0, 0.5),
  stringsAsFactors = FALSE
)

# Imprimir el DataFrame "datos_Asthma_Ig_hypers"
print(datos_Asthma_Ig_hypers)


  Variante Ranking Phenotypic_Score
1   PTGER2       1             0.93
2    TBX21       2             0.93
3     HNMT       8             0.50
4    ADRB2      16             0.00
5   PLA2G7      21             0.50


In [ ]:
# Crear el DataFrame "datos_Asthma_Abnormality_Igsystem"
datos_Asthma_Abnormality_Igsystem <- data.frame(
  Variante = c("PTGER2", "TBX21", "HNMT", "ADRB2", "PLA2G7"),
  Ranking = c(1, 2, 8, 16, 21),
  Phenotypic_Score = c(1, 1, 0.5, 0.0, 0.5),
  stringsAsFactors = FALSE
)

# Imprimir el DataFrame "datos_Asthma_Abnormality_Igsystem"
print(datos_Asthma_Abnormality_Igsystem)


  Variante Ranking Phenotypic_Score
1   PTGER2       1              1.0
2    TBX21       2              1.0
3     HNMT       8              0.5
4    ADRB2      16              0.0
5   PLA2G7      21              0.5


In [ ]:
# Crear el DataFrame "datos_Ab_endocrine_system"
datos_Ab_endocrine_system <- data.frame(
  Variante = c("PTGER2", "TBX21", "HNMT", "ADRB2", "PLA2G7"),
  Ranking = c(3, 8, 9, 17, 21),
  Phenotypic_Score = c(0.5, 0.5, 0.5, 0.0, 0.5),
  stringsAsFactors = FALSE
)

# Imprimir el DataFrame "Ab_endocrine_system"
print(datos_Ab_endocrine_system)


  Variante Ranking Phenotypic_Score
1   PTGER2       3              0.5
2    TBX21       8              0.5
3     HNMT       9              0.5
4    ADRB2      17              0.0
5   PLA2G7      21              0.5


In [ ]:
# Crear el DataFrame "datos_all_HPOS"
datos_all_HPOS <- data.frame(
  Variante = c("PTGER2", "TBX21", "HNMT", "ADRB2", "PLA2G7"),
  Ranking = c(4, 5, 10, 17, 21),
  Phenotypic_Score = c(0.67, 0.97, 0.5, 0.0, 0.5),
  stringsAsFactors = FALSE
)

# Imprimir el DataFrame "all_HPOS"
print(datos_all_HPOS)


  Variante Ranking Phenotypic_Score
1   PTGER2       4             0.67
2    TBX21       5             0.97
3     HNMT      10             0.50
4    ADRB2      17             0.00
5   PLA2G7      21             0.50


In [ ]:
# Crear el DataFrame "datos_Decreased_T4"
datos_Decreased_T4 <- data.frame(
  Variante = c("PTGER2", "TBX21", "HNMT", "ADRB2", "PLA2G7"),
  Ranking = c(5, 10, 9, 17, 21),
  Phenotypic_Score = c(0.5, 0.5, 0.5, 0.0, 0.5),
  stringsAsFactors = FALSE
)

# Imprimir el DataFrame "datos_Decreased_T4"
print(datos_Decreased_T4)


  Variante Ranking Phenotypic_Score
1   PTGER2       5              0.5
2    TBX21      10              0.5
3     HNMT       9              0.5
4    ADRB2      17              0.0
5   PLA2G7      21              0.5


COMPARACIONES ESTADISTICAS

In [ ]:
# Comparación de Rankings utilizando la prueba de Mann-Whitney
mw_ranking_results <- list()
dataframes <- list(datos_Sin_HPO, datos_Asthma, datos_Asthma_ARS, datos_Asthma_ARP, datos_Ab_Resp_system, datos_Asthma_Ig_hypers, datos_Asthma_Abnormality_Igsystem, datos_Ab_endocrine_system, datos_all_HPOS, datos_Decreased_T4)

# Asignar nombres a los dataframes
names(dataframes) <- c("datos_Sin_HPO", "datos_Asthma", "datos_Asthma_ARS", "datos_Asthma_ARP", "datos_Ab_Resp_system", "datos_Asthma_Ig_hypers", "datos_Asthma_Abnormality_Igsystem", "datos_Ab_endocrine_system", "datos_all_HPOS", "datos_Decreased_T4")

cat("Comparación de Rankings utilizando la prueba de Mann-Whitney:\n")
counter <- 1

for (i in 1:(length(dataframes) - 1)) {
  for (j in (i+1):length(dataframes)) {
    df1 <- dataframes[[i]]
    df2 <- dataframes[[j]]
    comparison <- try(wilcox.test(df1$Ranking, df2$Ranking, exact = FALSE), silent = TRUE)
    if (inherits(comparison, "try-error")) {
      cat(counter, ". No se pudo calcular el valor p para la comparación:", names(dataframes)[i], "con", names(dataframes)[j], "para el ranking usando Mann-Whitney.\n")
    } else {
      p_value <- comparison$p.value
      comparison_name <- paste(names(dataframes)[i], "con", names(dataframes)[j])
      mw_ranking_results[[comparison_name]] <- p_value
      cat(counter, ". El valor p de la comparación", names(dataframes)[i], "con", names(dataframes)[j], "para el ranking usando Mann-Whitney es:", p_value, "\n")
    }
    counter <- counter + 1
  }
}

# Comparación de Phenotypic Scores utilizando la prueba de Mann-Whitney
mw_score_results <- list()

cat("\nComparación de Phenotypic Scores utilizando la prueba de Mann-Whitney:\n")
counter <- 1

for (i in 1:(length(dataframes) - 1)) {
  for (j in (i+1):length(dataframes)) {
    df1 <- dataframes[[i]]
    df2 <- dataframes[[j]]
    comparison <- try(wilcox.test(df1$Phenotypic_Score, df2$Phenotypic_Score, exact = FALSE), silent = TRUE)
    if (inherits(comparison, "try-error")) {
      cat(counter, ". No se pudo calcular el valor p para la comparación:", names(dataframes)[i], "con", names(dataframes)[j], "para el Phenotypic_score usando Mann-Whitney.\n")
    } else {
      p_value <- comparison$p.value
      comparison_name <- paste(names(dataframes)[i], "con", names(dataframes)[j])
      mw_score_results[[comparison_name]] <- p_value
      cat(counter, ". El valor p de la comparación", names(dataframes)[i], "con", names(dataframes)[j], "para el Phenotypic_score usando Mann-Whitney es:", p_value, "\n")
    }
    counter <- counter + 1
  }
}

# Comparación de Rankings utilizando el test de Chi-cuadrado
chi2_ranking_results <- list()

cat("\nComparación de Rankings utilizando el test de Chi-cuadrado:\n")
counter <- 1

for (i in 1:(length(dataframes) - 1)) {
  for (j in (i+1):length(dataframes)) {
    df1 <- dataframes[[i]]
    df2 <- dataframes[[j]]
    contingency_table <- table(df1$Ranking, df2$Ranking)
    comparison <- try(chisq.test(contingency_table), silent = TRUE)
    if (inherits(comparison, "try-error")) {
      cat(counter, ". No se pudo calcular el valor p para la comparación:", names(dataframes)[i], "con", names(dataframes)[j], "para el ranking usando Chi2.\n")
    } else {
      p_value <- comparison$p.value
      comparison_name <- paste(names(dataframes)[i], "con", names(dataframes)[j])
      chi2_ranking_results[[comparison_name]] <- p_value
      cat(counter, ". El valor p de la comparación", names(dataframes)[i], "con", names(dataframes)[j], "para el ranking usando Chi2 es:", p_value, "\n")
    }
    counter <- counter + 1
  }
}

# Comparación de Phenotypic Scores utilizando el test de Chi-cuadrado
chi2_score_results <- list()

cat("\nComparación de Phenotypic Scores utilizando el test de Chi-cuadrado:\n")
counter <- 1

for (i in 1:(length(dataframes) - 1)) {
  for (j in (i+1):length(dataframes)) {
    df1 <- dataframes[[i]]
    df2 <- dataframes[[j]]
    contingency_table <- table(df1$Phenotypic_Score, df2$Phenotypic_Score)
    comparison <- try(chisq.test(contingency_table), silent = TRUE)
    if (inherits(comparison, "try-error")) {
      cat(counter, ". No se pudo calcular el valor p para la comparación:", names(dataframes)[i], "con", names(dataframes)[j], "para el Phenotypic_score usando Chi2.\n")
    } else {
      p_value <- comparison$p.value
      comparison_name <- paste(names(dataframes)[i], "con", names(dataframes)[j])
      chi2_score_results[[comparison_name]] <- p_value
      cat(counter, ". El valor p de la comparación", names(dataframes)[i], "con", names(dataframes)[j], "para el Phenotypic_score usando Chi2 es:", p_value, "\n")
    }
    counter <- counter + 1
  }
}


Comparación de Rankings utilizando la prueba de Mann-Whitney:
1 . El valor p de la comparación datos_Sin_HPO con datos_Asthma para el ranking usando Mann-Whitney es: 0.753298 
2 . El valor p de la comparación datos_Sin_HPO con datos_Asthma_ARS para el ranking usando Mann-Whitney es: 0.753298 
3 . El valor p de la comparación datos_Sin_HPO con datos_Asthma_ARP para el ranking usando Mann-Whitney es: 0.753298 
4 . El valor p de la comparación datos_Sin_HPO con datos_Ab_Resp_system para el ranking usando Mann-Whitney es: 0.753298 
5 . El valor p de la comparación datos_Sin_HPO con datos_Asthma_Ig_hypers para el ranking usando Mann-Whitney es: 0.753298 
6 . El valor p de la comparación datos_Sin_HPO con datos_Asthma_Abnormality_Igsystem para el ranking usando Mann-Whitney es: 0.753298 
7 . El valor p de la comparación datos_Sin_HPO con datos_Ab_endocrine_system para el ranking usando Mann-Whitney es: 1 
8 . El valor p de la comparación datos_Sin_HPO con datos_all_HPOS para el ranking usand

Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


1 . El valor p de la comparación datos_Sin_HPO con datos_Asthma para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


2 . El valor p de la comparación datos_Sin_HPO con datos_Asthma_ARS para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


3 . El valor p de la comparación datos_Sin_HPO con datos_Asthma_ARP para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


4 . El valor p de la comparación datos_Sin_HPO con datos_Ab_Resp_system para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


5 . El valor p de la comparación datos_Sin_HPO con datos_Asthma_Ig_hypers para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


6 . El valor p de la comparación datos_Sin_HPO con datos_Asthma_Abnormality_Igsystem para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


7 . El valor p de la comparación datos_Sin_HPO con datos_Ab_endocrine_system para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


8 . El valor p de la comparación datos_Sin_HPO con datos_all_HPOS para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


9 . El valor p de la comparación datos_Sin_HPO con datos_Decreased_T4 para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


10 . El valor p de la comparación datos_Asthma con datos_Asthma_ARS para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


11 . El valor p de la comparación datos_Asthma con datos_Asthma_ARP para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


12 . El valor p de la comparación datos_Asthma con datos_Ab_Resp_system para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


13 . El valor p de la comparación datos_Asthma con datos_Asthma_Ig_hypers para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


14 . El valor p de la comparación datos_Asthma con datos_Asthma_Abnormality_Igsystem para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


15 . El valor p de la comparación datos_Asthma con datos_Ab_endocrine_system para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


16 . El valor p de la comparación datos_Asthma con datos_all_HPOS para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


17 . El valor p de la comparación datos_Asthma con datos_Decreased_T4 para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


18 . El valor p de la comparación datos_Asthma_ARS con datos_Asthma_ARP para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


19 . El valor p de la comparación datos_Asthma_ARS con datos_Ab_Resp_system para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


20 . El valor p de la comparación datos_Asthma_ARS con datos_Asthma_Ig_hypers para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


21 . El valor p de la comparación datos_Asthma_ARS con datos_Asthma_Abnormality_Igsystem para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


22 . El valor p de la comparación datos_Asthma_ARS con datos_Ab_endocrine_system para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


23 . El valor p de la comparación datos_Asthma_ARS con datos_all_HPOS para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


24 . El valor p de la comparación datos_Asthma_ARS con datos_Decreased_T4 para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


25 . El valor p de la comparación datos_Asthma_ARP con datos_Ab_Resp_system para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


26 . El valor p de la comparación datos_Asthma_ARP con datos_Asthma_Ig_hypers para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


27 . El valor p de la comparación datos_Asthma_ARP con datos_Asthma_Abnormality_Igsystem para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


28 . El valor p de la comparación datos_Asthma_ARP con datos_Ab_endocrine_system para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


29 . El valor p de la comparación datos_Asthma_ARP con datos_all_HPOS para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


30 . El valor p de la comparación datos_Asthma_ARP con datos_Decreased_T4 para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


31 . El valor p de la comparación datos_Ab_Resp_system con datos_Asthma_Ig_hypers para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


32 . El valor p de la comparación datos_Ab_Resp_system con datos_Asthma_Abnormality_Igsystem para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


33 . El valor p de la comparación datos_Ab_Resp_system con datos_Ab_endocrine_system para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


34 . El valor p de la comparación datos_Ab_Resp_system con datos_all_HPOS para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


35 . El valor p de la comparación datos_Ab_Resp_system con datos_Decreased_T4 para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


36 . El valor p de la comparación datos_Asthma_Ig_hypers con datos_Asthma_Abnormality_Igsystem para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


37 . El valor p de la comparación datos_Asthma_Ig_hypers con datos_Ab_endocrine_system para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


38 . El valor p de la comparación datos_Asthma_Ig_hypers con datos_all_HPOS para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


39 . El valor p de la comparación datos_Asthma_Ig_hypers con datos_Decreased_T4 para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


40 . El valor p de la comparación datos_Asthma_Abnormality_Igsystem con datos_Ab_endocrine_system para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


41 . El valor p de la comparación datos_Asthma_Abnormality_Igsystem con datos_all_HPOS para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


42 . El valor p de la comparación datos_Asthma_Abnormality_Igsystem con datos_Decreased_T4 para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


43 . El valor p de la comparación datos_Ab_endocrine_system con datos_all_HPOS para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


44 . El valor p de la comparación datos_Ab_endocrine_system con datos_Decreased_T4 para el ranking usando Chi2 es: 0.2202206 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


45 . El valor p de la comparación datos_all_HPOS con datos_Decreased_T4 para el ranking usando Chi2 es: 0.2202206 

Comparación de Phenotypic Scores utilizando el test de Chi-cuadrado:


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


1 . El valor p de la comparación datos_Sin_HPO con datos_Asthma para el Phenotypic_score usando Chi2 es: 0.8187308 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


2 . El valor p de la comparación datos_Sin_HPO con datos_Asthma_ARS para el Phenotypic_score usando Chi2 es: 0.8187308 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


3 . El valor p de la comparación datos_Sin_HPO con datos_Asthma_ARP para el Phenotypic_score usando Chi2 es: 0.8187308 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


4 . El valor p de la comparación datos_Sin_HPO con datos_Ab_Resp_system para el Phenotypic_score usando Chi2 es: 0.8187308 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


5 . El valor p de la comparación datos_Sin_HPO con datos_Asthma_Ig_hypers para el Phenotypic_score usando Chi2 es: 0.8187308 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


6 . El valor p de la comparación datos_Sin_HPO con datos_Asthma_Abnormality_Igsystem para el Phenotypic_score usando Chi2 es: 0.8187308 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


7 . El valor p de la comparación datos_Sin_HPO con datos_Ab_endocrine_system para el Phenotypic_score usando Chi2 es: 0.1797125 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


8 . El valor p de la comparación datos_Sin_HPO con datos_all_HPOS para el Phenotypic_score usando Chi2 es: 0.8964324 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


9 . El valor p de la comparación datos_Sin_HPO con datos_Decreased_T4 para el Phenotypic_score usando Chi2 es: 0.1797125 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


10 . El valor p de la comparación datos_Asthma con datos_Asthma_ARS para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


11 . El valor p de la comparación datos_Asthma con datos_Asthma_ARP para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


12 . El valor p de la comparación datos_Asthma con datos_Ab_Resp_system para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


13 . El valor p de la comparación datos_Asthma con datos_Asthma_Ig_hypers para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


14 . El valor p de la comparación datos_Asthma con datos_Asthma_Abnormality_Igsystem para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


15 . El valor p de la comparación datos_Asthma con datos_Ab_endocrine_system para el Phenotypic_score usando Chi2 es: 0.082085 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


16 . El valor p de la comparación datos_Asthma con datos_all_HPOS para el Phenotypic_score usando Chi2 es: 0.124652 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


17 . El valor p de la comparación datos_Asthma con datos_Decreased_T4 para el Phenotypic_score usando Chi2 es: 0.082085 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


18 . El valor p de la comparación datos_Asthma_ARS con datos_Asthma_ARP para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


19 . El valor p de la comparación datos_Asthma_ARS con datos_Ab_Resp_system para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


20 . El valor p de la comparación datos_Asthma_ARS con datos_Asthma_Ig_hypers para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


21 . El valor p de la comparación datos_Asthma_ARS con datos_Asthma_Abnormality_Igsystem para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


22 . El valor p de la comparación datos_Asthma_ARS con datos_Ab_endocrine_system para el Phenotypic_score usando Chi2 es: 0.082085 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


23 . El valor p de la comparación datos_Asthma_ARS con datos_all_HPOS para el Phenotypic_score usando Chi2 es: 0.124652 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


24 . El valor p de la comparación datos_Asthma_ARS con datos_Decreased_T4 para el Phenotypic_score usando Chi2 es: 0.082085 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


25 . El valor p de la comparación datos_Asthma_ARP con datos_Ab_Resp_system para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


26 . El valor p de la comparación datos_Asthma_ARP con datos_Asthma_Ig_hypers para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


27 . El valor p de la comparación datos_Asthma_ARP con datos_Asthma_Abnormality_Igsystem para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


28 . El valor p de la comparación datos_Asthma_ARP con datos_Ab_endocrine_system para el Phenotypic_score usando Chi2 es: 0.082085 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


29 . El valor p de la comparación datos_Asthma_ARP con datos_all_HPOS para el Phenotypic_score usando Chi2 es: 0.124652 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


30 . El valor p de la comparación datos_Asthma_ARP con datos_Decreased_T4 para el Phenotypic_score usando Chi2 es: 0.082085 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


31 . El valor p de la comparación datos_Ab_Resp_system con datos_Asthma_Ig_hypers para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


32 . El valor p de la comparación datos_Ab_Resp_system con datos_Asthma_Abnormality_Igsystem para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


33 . El valor p de la comparación datos_Ab_Resp_system con datos_Ab_endocrine_system para el Phenotypic_score usando Chi2 es: 0.082085 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


34 . El valor p de la comparación datos_Ab_Resp_system con datos_all_HPOS para el Phenotypic_score usando Chi2 es: 0.124652 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


35 . El valor p de la comparación datos_Ab_Resp_system con datos_Decreased_T4 para el Phenotypic_score usando Chi2 es: 0.082085 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


36 . El valor p de la comparación datos_Asthma_Ig_hypers con datos_Asthma_Abnormality_Igsystem para el Phenotypic_score usando Chi2 es: 0.04042768 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


37 . El valor p de la comparación datos_Asthma_Ig_hypers con datos_Ab_endocrine_system para el Phenotypic_score usando Chi2 es: 0.082085 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


38 . El valor p de la comparación datos_Asthma_Ig_hypers con datos_all_HPOS para el Phenotypic_score usando Chi2 es: 0.124652 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


39 . El valor p de la comparación datos_Asthma_Ig_hypers con datos_Decreased_T4 para el Phenotypic_score usando Chi2 es: 0.082085 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


40 . El valor p de la comparación datos_Asthma_Abnormality_Igsystem con datos_Ab_endocrine_system para el Phenotypic_score usando Chi2 es: 0.082085 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


41 . El valor p de la comparación datos_Asthma_Abnormality_Igsystem con datos_all_HPOS para el Phenotypic_score usando Chi2 es: 0.124652 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


42 . El valor p de la comparación datos_Asthma_Abnormality_Igsystem con datos_Decreased_T4 para el Phenotypic_score usando Chi2 es: 0.082085 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


43 . El valor p de la comparación datos_Ab_endocrine_system con datos_all_HPOS para el Phenotypic_score usando Chi2 es: 0.1717971 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


44 . El valor p de la comparación datos_Ab_endocrine_system con datos_Decreased_T4 para el Phenotypic_score usando Chi2 es: 0.4017356 


Warning message in chisq.test(contingency_table):
“Chi-squared approximation may be incorrect”


45 . El valor p de la comparación datos_all_HPOS con datos_Decreased_T4 para el Phenotypic_score usando Chi2 es: 0.1717971 


In [ ]:
# Filtrar valores p menores a 0.05 para la comparación de Rankings utilizando Mann-Whitney
significant_ranking_results <- Filter(function(x) x < 0.05, unlist(mw_ranking_results))
cat("\nComparaciones de Rankings con diferencias estadísticamente significativas (Mann-Whitney):\n")
cat(paste(names(significant_ranking_results), "(", significant_ranking_results, ")", sep = " "), "\n")

# Filtrar valores p menores a 0.05 para la comparación de Phenotypic Scores utilizando Mann-Whitney
significant_score_results <- Filter(function(x) x < 0.05, unlist(mw_score_results))
cat("\nComparaciones de Phenotypic Scores con diferencias estadísticamente significativas (Mann-Whitney):\n")
cat(paste(names(significant_score_results), "(", significant_score_results, ")", sep = " "), "\n")

# Filtrar valores p menores a 0.05 para la comparación de Rankings utilizando Chi-cuadrado
significant_chi2_ranking_results <- Filter(function(x) x < 0.05, unlist(chi2_ranking_results))
cat("\nComparaciones de Rankings con diferencias estadísticamente significativas (Chi-cuadrado):\n")
cat(paste(names(significant_chi2_ranking_results), "(", significant_chi2_ranking_results, ")", sep = " "), "\n")

# Filtrar valores p menores a 0.05 para la comparación de Phenotypic Scores utilizando Chi-cuadrado
significant_chi2_score_results <- Filter(function(x) x < 0.05, unlist(chi2_score_results))
cat("\nComparaciones de Phenotypic Scores con diferencias estadísticamente significativas (Chi-cuadrado):\n")
cat(paste(names(significant_chi2_score_results), "(", significant_chi2_score_results, ")", sep = " "), "\n")



Comparaciones de Rankings con diferencias estadísticamente significativas (Mann-Whitney):
 (  ) 

Comparaciones de Phenotypic Scores con diferencias estadísticamente significativas (Mann-Whitney):
datos_Sin_HPO con datos_Asthma ( 0.0242514051837674 ) datos_Sin_HPO con datos_Asthma_ARS ( 0.0242514051837674 ) datos_Sin_HPO con datos_Asthma_ARP ( 0.0242514051837674 ) datos_Sin_HPO con datos_Ab_Resp_system ( 0.0242514051837674 ) datos_Sin_HPO con datos_Asthma_Ig_hypers ( 0.0242514051837674 ) datos_Sin_HPO con datos_Asthma_Abnormality_Igsystem ( 0.0242514051837674 ) datos_Sin_HPO con datos_Ab_endocrine_system ( 0.0199644533052161 ) datos_Sin_HPO con datos_all_HPOS ( 0.0248084245478095 ) datos_Sin_HPO con datos_Decreased_T4 ( 0.0199644533052161 ) 

Comparaciones de Rankings con diferencias estadísticamente significativas (Chi-cuadrado):
 (  ) 

Comparaciones de Phenotypic Scores con diferencias estadísticamente significativas (Chi-cuadrado):
datos_Asthma con datos_Asthma_ARS ( 0.04042768199

In [ ]:
# Combinar los dataframes en uno solo excluyendo
datos <- rbind(datos_Sin_HPO, datos_Asthma, datos_Asthma_ARS, datos_Asthma_ARP, datos_Ab_Resp_system, datos_Asthma_Ig_hypers, datos_Asthma_Abnormality_Igsystem, datos_Ab_endocrine_system, datos_all_HPOS, datos_Decreased_T4)

# Ajuste de un modelo lineal
modelo <- lm(Ranking ~ Phenotypic_Score, data = datos)

print(summary(modelo))


Call:
lm(formula = Ranking ~ Phenotypic_Score, data = datos)

Residuals:
    Min      1Q  Median      3Q     Max 
-13.617  -2.845  -1.845   1.230  10.769 

Coefficients:
                 Estimate Std. Error t value Pr(>|t|)    
(Intercept)        16.618      1.423  11.677 1.25e-15 ***
Phenotypic_Score  -12.772      2.351  -5.432 1.83e-06 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 5.978 on 48 degrees of freedom
Multiple R-squared:  0.3807,	Adjusted R-squared:  0.3678 
F-statistic:  29.5 on 1 and 48 DF,  p-value: 1.828e-06




    Coeficientes: El coeficiente para la variable "Phenotypic_Score" es -12.772. Esto significa que, en promedio, un aumento de una unidad en el "Phenotypic_Score" se asocia con una disminución de aproximadamente 12.772 en el "Ranking", teniendo en cuenta el otro predictor constante.

    P-values: El p-value asociado al coeficiente de "Phenotypic_Score" es muy pequeño (1.83e-06), lo que indica una significancia estadística. Esto sugiere que hay una relación significativa entre el "Phenotypic_Score" y el "Ranking".

    R cuadrado: El valor R cuadrado es 0.3807, lo que significa que aproximadamente el 38.07% de la variabilidad en el "Ranking" se explica por la variable "Phenotypic_Score" en el modelo. El R cuadrado ajustado, que tiene en cuenta el número de predictores y los grados de libertad, es 0.3678.

    Error estándar residual: El error estándar residual es 5.978, lo que sugiere la dispersión típica de las observaciones alrededor de la línea de regresión.

    F-statistic: El valor de F-statistic es 29.5, con un p-value asociado muy pequeño (1.828e-06). Esto indica que el modelo en su conjunto es estadísticamente significativo.

    el modelo de regresión lineal muestra que el "Phenotypic_Score" tiene un efecto significativo en el "Ranking", y explica aproximadamente el 38.07% de la variabilidad observada.
    
    Dado que el coeficiente para la variable "Phenotypic_Score" es negativo (-12.772), esto implica que un aumento de una unidad en el "Phenotypic_Score" se asocia con una disminución de aproximadamente 12.772 en el "Ranking", lo cual significa que el "Ranking" se acerca a 1 (mejor posición) cuando el "Phenotypic_Score" aumenta

In [ ]:
# Combinar los dataframes en uno solo
datos <- rbind(datos_Sin_HPO, datos_Asthma, datos_Asthma_ARS, datos_Asthma_ARP, datos_Ab_Resp_system, datos_Asthma_Ig_hypers, datos_Asthma_Abnormality_Igsystem, datos_Ab_endocrine_system, datos_all_HPOS, datos_Decreased_T4)

# Ajuste de un modelo lineal
modelo <- lm(Ranking ~ Estrategia, data = datos)

# Resumen del modelo
summary(modelo)



Call:
lm(formula = Ranking ~ Estrategia, data = datos)

Residuals:
   Min     1Q Median     3Q    Max 
 -9.00  -7.55  -1.80   7.60  11.40 

Coefficients:
                                      Estimate Std. Error t value Pr(>|t|)   
(Intercept)                             11.600      3.691   3.143  0.00314 **
EstrategiaAb_Resp_system                -2.000      5.219  -0.383  0.70360   
Estrategiaall_HPOS                      -0.200      5.219  -0.038  0.96962   
EstrategiaAsthma                        -1.600      5.219  -0.307  0.76077   
EstrategiaAsthma_Abnormality_Igsystem   -2.000      5.219  -0.383  0.70360   
EstrategiaAsthma_ARP                    -1.600      5.219  -0.307  0.76077   
EstrategiaAsthma_ARS                    -1.600      5.219  -0.307  0.76077   
EstrategiaAsthma_Ig_hypers              -2.000      5.219  -0.383  0.70360   
EstrategiaDecreased_T4                   0.800      5.219   0.153  0.87895   
EstrategiaSin_HPO                       -1.800      5.219  -0.345

In [ ]:
# Combinar los dataframes en uno solo
datos <- rbind(datos_Sin_HPO, datos_Asthma, datos_Asthma_ARS, datos_Asthma_ARP, datos_Ab_Resp_system, datos_Asthma_Ig_hypers, datos_Asthma_Abnormality_Igsystem, datos_Ab_endocrine_system, datos_all_HPOS, datos_Decreased_T4)

# Ajuste de un modelo lineal
modelo <- lm(Phenotypic_Score ~ Estrategia, data = datos)

# Resumen del modelo
summary(modelo)




Call:
lm(formula = Phenotypic_Score ~ Estrategia, data = datos)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.6000 -0.1000  0.0000  0.2995  0.4420 

Coefficients:
                                        Estimate Std. Error t value Pr(>|t|)  
(Intercept)                            4.000e-01  1.560e-01   2.564   0.0142 *
EstrategiaAb_Resp_system               2.000e-01  2.207e-01   0.906   0.3702  
Estrategiaall_HPOS                     1.280e-01  2.207e-01   0.580   0.5651  
EstrategiaAsthma                       1.680e-01  2.207e-01   0.761   0.4509  
EstrategiaAsthma_Abnormality_Igsystem  2.000e-01  2.207e-01   0.906   0.3702  
EstrategiaAsthma_ARP                   2.000e-01  2.207e-01   0.906   0.3702  
EstrategiaAsthma_ARS                   2.000e-01  2.207e-01   0.906   0.3702  
EstrategiaAsthma_Ig_hypers             1.720e-01  2.207e-01   0.779   0.4403  
EstrategiaDecreased_T4                -6.081e-17  2.207e-01   0.000   1.0000  
EstrategiaSin_HPO                   

In [ ]:
# Combinar los dataframes en uno solo
datos <- rbind(datos_Sin_HPO, datos_Asthma, datos_Asthma_ARS, datos_Asthma_ARP, datos_Ab_Resp_system, datos_Asthma_Ig_hypers, datos_Asthma_Abnormality_Igsystem, datos_Ab_endocrine_system, datos_all_HPOS, datos_Decreased_T4)

# Ajuste de un modelo lineal para Phenotypic_Score con ranking como variable predictora
modelo <- lm(Phenotypic_Score ~ Ranking + Estrategia, data = datos)

# Resumen del modelo
summary(modelo)



Call:
lm(formula = Phenotypic_Score ~ Ranking + Estrategia, data = datos)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.40976 -0.15850  0.02569  0.17111  0.37942 

Coefficients:
                                       Estimate Std. Error t value Pr(>|t|)    
(Intercept)                            0.744815   0.125488   5.935 6.37e-07 ***
Ranking                               -0.029725   0.004815  -6.174 2.97e-07 ***
EstrategiaAb_Resp_system               0.140549   0.159215   0.883  0.38277    
Estrategiaall_HPOS                     0.122055   0.158926   0.768  0.44711    
EstrategiaAsthma                       0.120439   0.159110   0.757  0.45362    
EstrategiaAsthma_Abnormality_Igsystem  0.140549   0.159215   0.883  0.38277    
EstrategiaAsthma_ARP                   0.152439   0.159110   0.958  0.34393    
EstrategiaAsthma_ARS                   0.152439   0.159110   0.958  0.34393    
EstrategiaAsthma_Ig_hypers             0.112549   0.159215   0.707  0.48383    
Estrat

In [ ]:
# Combinar los dataframes en uno solo
datos <- rbind(datos_Sin_HPO, datos_Asthma, datos_Asthma_ARS, datos_Asthma_ARP, datos_Ab_Resp_system, datos_Asthma_Ig_hypers, datos_Asthma_Abnormality_Igsystem, datos_Ab_endocrine_system, datos_all_HPOS, datos_Decreased_T4)

# Ajuste de un modelo lineal para Ranking con Phenotypic_Score como variable predictora
modelo <- lm(Ranking ~ Phenotypic_Score + Estrategia, data = datos)

# Resumen del modelo
summary(modelo)


Call:
lm(formula = Ranking ~ Phenotypic_Score + Estrategia, data = datos)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.9371 -3.1427 -1.9428 -0.0343 11.2000 

Coefficients:
                                      Estimate Std. Error t value Pr(>|t|)    
(Intercept)                            18.2514     2.8679   6.364 1.62e-07 ***
Phenotypic_Score                      -16.6286     2.6933  -6.174 2.97e-07 ***
EstrategiaAb_Resp_system                1.3257     3.7972   0.349   0.7289    
Estrategiaall_HPOS                      1.9285     3.7746   0.511   0.6123    
EstrategiaAsthma                        1.1936     3.7860   0.315   0.7542    
EstrategiaAsthma_Abnormality_Igsystem   1.3257     3.7972   0.349   0.7289    
EstrategiaAsthma_ARP                    1.7257     3.7972   0.454   0.6520    
EstrategiaAsthma_ARS                    1.7257     3.7972   0.454   0.6520    
EstrategiaAsthma_Ig_hypers              0.8601     3.7873   0.227   0.8215    
EstrategiaDecreased_T4    